test/e2e/webhook_test.go

```diff
@@ -0,0 +1,54 @@
+package e2e
+
+import (
+	"strconv"
+
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+
+	nmstatev1alpha1 "github.com/nmstate/kubernetes-nmstate/pkg/apis/nmstate/v1alpha1"
+	mutatingwebhook "github.com/nmstate/kubernetes-nmstate/pkg/webhook/mutating"
+)
+
+var _ = Describe("Mutating Admission Webhook", func() {
+	policyName := "webhook-test"
+	Context("when policy is created", func() {
+		BeforeEach(func() {
+			setDesiredStateWithPolicy(policyName, linuxBrUp(bridge1))
+			waitForAvailablePolicy(policyName)
+		})
+		AfterEach(func() {
+			setDesiredStateWithPolicy(policyName, linuxBrAbsent(bridge1))
+			waitForAvailablePolicy(policyName)
+			deletePolicy(policyName)
+		})
+
+		It("should add an annotation with mutation timestamp", func() {
+			policy := nodeNetworkConfigurationPolicy(policyName)
+			Expect(policy.ObjectMeta.Annotations).To(HaveKey(mutatingwebhook.TimestampLabelKey))
+		})
+		Context("and we updated it", func() {
+			var (
+				oldPolicy nmstatev1alpha1.NodeNetworkConfigurationPolicy
+			)
+			BeforeEach(func() {
+				oldPolicy = nodeNetworkConfigurationPolicy(policyName)
+				setDesiredStateWithPolicy(policyName, linuxBrAbsent(bridge1))
+				waitForAvailablePolicy(policyName)
```

Comment:
Now when this is really reliable, don't we have any sleeps or blocking in the rest of the tests we want to drop?

Reply:
What do you mean ?